lib/string.flow

```diff
@@ -512,6 +514,35 @@ doStrReplace2(prefix : string, suffix : string, old : string, new : string) -> s
 	}
 }
 
+strIReplace(s : string, old : string, new : string) -> string {
+	if (old == "") s
+	else if (strlen(s) > 500) list2string(doStrIReplace(s, 0, toLowerCase(old), new, makeList()))
+	else doStrIReplace2("", s, toLowerCase(old), new);
+}
+
+doStrIReplace(s : string, i : int, old : string, new : string, acc : List<string>) -> List<string> {
+	n = strlen(s);
+	if (i < n) {
+		hit = strRangeIndexOf(toLowerCase(s), old, i, n);
+		if (hit >= 0) {
+			before = substring(s, i, hit - i);
+			nacc = Cons(new, Cons(before, acc));
+			doStrIReplace(s, hit + strlen(old), old, new, nacc)
+		} else Cons(strRight(s, i), acc);
+	} else acc;
+}
+
+doStrIReplace2(prefix : string, suffix : string, old : string, new : string) -> string {
```

Comment:
intead of adding new it's better to extend current doStrReplace2 with passing there fn like toLowerCase or idfn. the same for doStrIReplace.
And also check if there are native sibling at haxe StringTools